### PR TITLE
IRF water take

### PR DESCRIPTION
**Pull request by nmizukami**
Make water abstraction and injection consistent with the other routine methods.

No answer change with `is_flux_wm` off

### CODE REVIEW

**Review by ekluzek — CHANGES_REQUESTED** [review]
Just a couple really small questions and requests...

File: route/build/src/irf_route.f90 [hunk 1]
```diff
@@ -58,8 +58,9 @@ SUBROUTINE irf_rch(this,         & ! irf_route_rch object to bound this procedur
  ! Local variables
  logical(lgt)                             :: verbose        ! check details of variables
  real(dp)                                 :: q_upstream     ! total discharge at top of the reach being processed
+ real(dp)                                 :: q_upstream_mod ! total discharge at top of the reach being processed
```
[the remark this answers is not in the document]
Comment (nmizukami) [reply route/build/src/irf_route.f90:61]: Comments corrected.

File: route/build/src/irf_route.f90 [hunk 2]
```diff
@@ -93,11 +94,47 @@ SUBROUTINE irf_rch(this,         & ! irf_route_rch object to bound this procedur
...
+  ! Water management - water injection or abstraction (irrigation or industrial/domestic water usage)
+  ! For water abstraction, water is extracted from the following priorities:
+  ! 1. existing storage(REACH_VOL(0), 2. upstream inflow , 3 lateral flow (BASIN_QR)
+  if((RCHFLX_out(iens,segIndex)%REACH_WM_FLUX /= realMissing).and.(is_flux_wm)) then
+    if (Qabs > 0) then ! positive == abstraction
+      if (RCHFLX_out(iens,segIndex)%ROUTE(idxIRF)%REACH_VOL(1)/dt > Qabs) then ! take out all abstraction from strorage
+        RCHFLX_out(iens,segIndex)%ROUTE(idxIRF)%REACH_VOL(1) = RCHFLX_out(iens,segIndex)%ROUTE(idxIRF)%REACH_VOL(1) - Qabs*dt
+      else ! if inital abstraction is greater than volume
+        Qabs = Qabs - RCHFLX_out(iens,segIndex)%ROUTE(idxIRF)%REACH_VOL(1)/dt ! get residual Qabs after extracting from strorage
+        RCHFLX_out(iens,segIndex)%ROUTE(idxIRF)%REACH_VOL(1) = 0._dp ! voluem gets 0
```
[the remark this answers is not in the document]
Comment (nmizukami) [reply route/build/src/irf_route.f90:114]: typo corrected

**Comment by nmizukami** [comment]
thank you Erik,   comments should be fixed now.

**Review by ekluzek — APPROVED** [review]

**Pull request merged**